READMEs/[[...slug]].md

```diff
@@ -0,0 +1,107 @@
+
+## The problem
+
+But this isn't the reality of our use case. Our pages are not pre-determined. Rather, our pages are determined by content that can be added/edited/removed by editors in our CMS, which means that _our_ build process needs to first become aware of the paths that it needs to build.
+
+## Pages Router: Our Dynamic Approach
+
+Enter dynamic routes. In our case, we use a single dynamic route as a catch-all for generating all the pages. In the case of dynamic routes, Next.js requires the implementation of a function `getStaticPaths`, which is called during a static-site build to calculate of all the paths for all the pages that need to be built by the template in question. Here's what happens during our build:
+
+1. When a static build is initiated, Next.js (as we saw previously) steps through each of the files in the `pages` directory. In this case, notably, it comes upon `[[..slug]].tsx`.
+
+2. In that file, Next.js finds and calls the `getStaticPaths` function.
+
+3. Inside that function, we ping the CMS for all the content it knows about for _each_ of the content types that we want to build.
+
+4. The entities found for each content type are handled slightly differently. Here are some examples:
+
+   - Most news stories map directly to a single path.
+   - Lovell Federal news stories need to be converted to two paths, one for VA and one for TRICARE (see: `READMEs/lovell.md`)
+   - Story Listing Pages need to have an additional path generated for each subsequent page that needs to be statically built (see: `READMEs/pagination.md`)
+
+5. Once all paths are determined for each content type, they are merged together into one large collection.
+
+6. `getStaticProps` is called for each and every entry in the path collection.
+
+7. The data returned from `getStaticProps` is passed to the template to generate an individual page (HTML) for each path.
+
+### So what are the brackets and the ellipsis?
+
+Good question! It seems like we could name this file whatever we want. And that would work just fine if Next.js were _only_ a static-site generator. But it's not. Consider this:
+
+When we are developing features, rarely would we want to wait for an entire build to finish to see our changes. That would be a painfully slow development process. Instead, by runing `next dev` (`yarn dev`) we can run a development server to build individual pages on demand. But, in order to make this possible, Next.js needs to know which page template to execute. This is where the file naming comes into play.
+
+Let's return to our simple routing example of a site with three pre-determined pages. When building this site, let's say we want to work on the `/about` page. So, we run `yarn dev` and then visit `localhost:3000/about`. When the Next.js development server receives the request for this page, the first thing it will look for is a file in the `pages` directory named `about.tsx`. Since we have one, Next.js will use that file to build the page.
```

Comment:
lets update all the `localhost:3000` to `localhost:3999`, it is the default in `.env.example` now